### PR TITLE
Update dvcyaml-files.md

### PR DESCRIPTION
**Pull request by jorgeorpinel**
per https://github.com/iterative/dvc.org/pull/4010#pullrequestreview-1130866784

### CODE REVIEW

**Review by shcheklein — APPROVED** [review]

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 1]
```diff
@@ -499,7 +499,8 @@ stages:
 <admon type="tip">
 
 Both individual foreach stages (`train@1`) and groups of foreach stages
-(`train`) may be used in commands that accept stage targets.
+(`train`) may be used in commands that accept stage targets such as
+`dvc repro` and `dvc stage list`.
 
 </admon>
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:505]: Wait though... Shouldn't this note be much higher up after the example with foreach stages named `train`? Currently it's under a seemingly unrelated code bloc @dberenbaum 👍🏼 

image

> See https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#foreach-stages

Comment (daavoo) [reply content/docs/user-guide/project-structure/dvcyaml-files.md:505]: Or maybe just one block up before the `mystages` code block and modify the text to include example of syntax for dictionary:

Both individual foreach stages (`build@uk` / `train@1`) and groups of foreach stages (`build` / `train`)

Comment (dberenbaum) [reply content/docs/user-guide/project-structure/dvcyaml-files.md:505]: Wherever you think is best @jorgeorpinel

**Review by dberenbaum — APPROVED** [review]

File: content/docs/user-guide/project-structure/dvcyaml-files.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/dvcyaml-files.md]: Updated and moved.

File: content/docs/command-reference/plots/diff.md [hunk 2]
```diff
@@ -48,7 +48,6 @@ Another way to display plots is the `dvc plots show` command, which just lists
 all the current plots, without comparisons.
 
 [certain metrics]: /doc/user-guide/visualizing-plots#supported-plot-file-formats
-
 [plot templates]:
```
Comment (jorgeorpinel) [inline content/docs/command-reference/plots/diff.md:51]: The linter fixed this during a merge of `main` (to solve conflicts). Merging in...
